lib/autoPipelining.ts

```diff
@@ -104,7 +104,15 @@ export function executeWithAutoPipelining(
   }
 
   // If we have slot information, we can improve routing by grouping slots served by the same subset of nodes
-  const slotKey = client.isCluster ? client.slots[calculateSlot(args[0])].join(",") : 'main';
+  const slotKey = client.isCluster
+    ? client.slots[
+        calculateSlot(
+          client.options.keyPrefix
+            ? `${client.options.keyPrefix}${args[0]}`
+            : args[0]
+        )
+      ].join(",")
+    : "main";
```

Comment:
```suggestion
  const prefix = client.options.keyPrefix || '';
  const slotKey = client.isCluster
    ? client.slots[
        calculateSlot(`${prefix}${args[0]}`)
      ].join(",")
    : "main";
```

this should likely be cheaper. we can also avoid assigning empty string to keyPrefix if we set it in the constructor

Reply:
`redis.get([], [], ['key'])` is technically allowed by ioredis because all of the args get passed to lodash flatten. I added getFirstValueInFlattenedArray and some unit tests to work as a fallback for more common cases.

Less common cases (_.isArguments) may benefit from https://github.com/lodash/lodash/blob/master/.internal/isFlattenable.js but that seems brittle, and I don't know ioredis's policy on using internal implementation details like that

Reply:
Did something similar, but accounting for the fact that `args[0]` can be an array